src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3936,21 +3940,87 @@ private static string MatchCharacterClass(bool hasTextInfo, RegexOptions options
                 }
             }
 
-            // All options after this point require a ch local.
-            additionalDeclarations.Add("char ch;");
+            // Next, handle simple sets of two ranges where a simple bit manipulation transforms one into the other, e.g. [A-Za-z].
+            // This can be implemented as if it were a single range, with an additional bitwise operation.
+            if (!invariant &&
+                !RegexCharClass.IsNegated(charClass) &&
+                RegexCharClass.TryGetDoubleRange(charClass, out (char LowInclusive, char HighInclusive) rangeUnmasked, out (char LowInclusive, char HighInclusive) rangeMasked) &&
+                rangeUnmasked.LowInclusive != rangeUnmasked.HighInclusive && // more than a single char in range0 (just to simplify, since single chars should have already been handled)
+                rangeMasked.LowInclusive != rangeMasked.HighInclusive && // more than a single char in range1 (just to simplify, since single chars should have already been handled)
+                RegexCharClass.DifferByOneBit(rangeUnmasked.LowInclusive, rangeMasked.LowInclusive, out int lowMask) &&
```

Comment:
Is there no case where the Low and high inclusive ends of the ranges might have the same mask but some character in the middle of the range have a different mask? I guess my question is if it is safe to just check that the masks of the beginning and end of the ranges and assume all characters in between have the same mask.

Reply:
It's making my head hurt to try to answer that, so since really the only thing that matters are ASCII letters, I've just narrowed this down to handling that.

Reply:
FWIW, I *think* your assumption seemed right, but I believe it wouldn’t be trivial to prove this so my question was mostly me wondering whether you knew this to be true.